From 4f3b35c58ffb755cd7371faa44e2914fd7be3f0a Mon Sep 17 00:00:00 2001
From: Bernd Kuhls <bernd.kuhls@t-online.de>
Date: Mon, 4 Jan 2021 22:32:28 +0100
Subject: [PATCH] [linux] fix drm object build with gcc-4.9

Fixes build error introduced by PR18858:

kodi/xbmc/windowing/gbm/drm/DRMObject.cpp: In member function 'std::tuple<bool, long unsigned int> KODI::WINDOWING::GBM::CDRMObject::GetPropertyValue(const string&, const string&) const':
kodi/xbmc/windowing/gbm/drm/DRMObject.cpp:103:39: error: converting to 'std::tuple<bool, long unsigned int>' from initializer list would use explicit constructor 'constexpr std::tuple<_T1, _T2>::tuple(_U1&&, _U2&&) [with _U1 = bool; _U2 = long long unsigned int&; <template-parameter-2-3> = void; _T1 = bool; _T2 = long unsigned int]'
     return {true, prop->enums[j].value};

Patch sent upstream: https://github.com/xbmc/xbmc/pull/19020

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>
---
 xbmc/windowing/gbm/drm/DRMObject.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/windowing/gbm/drm/DRMObject.cpp b/xbmc/windowing/gbm/drm/DRMObject.cpp
index 7d27c6f059..f93f014f7d 100644
--- a/xbmc/windowing/gbm/drm/DRMObject.cpp
+++ b/xbmc/windowing/gbm/drm/DRMObject.cpp
@@ -100,7 +100,7 @@ std::tuple<bool, uint64_t> CDRMObject::GetPropertyValue(const std::string& name,
     if (prop->enums[j].name != valueName)
       continue;
 
-    return {true, prop->enums[j].value};
+    return std::make_tuple(true, prop->enums[j].value);
   }
 
   return std::make_tuple(false, 0);
-- 
2.29.2

